airflow_connector.py

```diff
@@ -0,0 +1,38 @@
+"""Connects to the Cloud SQL database."""
+
+
+class AirflowDB(object):
+  """"Provides the methods which allow interaction with the Airflow SQL database."""  # pylint: disable=line-too-long
+
+  def __init__(self, db_connection):
+    """Connects to the Cloud SQL database."""
+    # If I try to put this in main.py, then I get a circular import error when
+    # I try to import the connection/AiflowDB object into resources.py
+    # I could put it somewhere else, but for now it's here. It also seems that
+    # App Engine simply let me delete the env vars and everything is working finished
+
+    self._db = db_connection
```

Comment:
Should the connection be made inside instead?

Reply:
What do you mean inside? Inside what?

Reply:
Creating and initialing db connection inside this class, instead of taking a connection created outside.

This looks like the abstraction to handle all DB related stuff.

Reply:
That was what I had initially, but Seb asked me to change it [here](https://github.com/istio-releases/release-ui/pull/31#discussion_r201759439)

Reply:
@sebastienvas I suppose the airflow connector should abstract out all DB/airflow interactions, including dealing with the DB. What are the reasons why you want the connection object to be passed instead and have main create the connection?

I suppose this abstraction should have all the required info (i.e. user/pass/connection_info/etc) to be able to create connections. Also, connection object can turn bad due to network glitch (network resiliency is not always implemented in driver) and this abstraction may need to re-create db connection.

Not sure what "env vars" was referred to in your comment, but I agree that we should not rely on env vars here for sure. This can just take all required info as input params.

Reply:
>Also, connection object can turn bad due to network glitch (network resiliency is not always implemented in driver) and this abstraction may need to re-create db connection.
I've actually run into this problem a few times already during development/testing